src/main/java/at/hannibal2/skyhanni/features/inventory/SkyblockGuideHighlightFeature.kt

```diff
@@ -0,0 +1,159 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.events.LorenzToolTipEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.name
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.StringUtils.anyMatches
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.intellij.lang.annotations.Language
+
+val group = RepoPattern.group("skyblockguide.highlight")
+
+private const val keyPrefixInventory = "inventory"
+private const val keyPrefixCondition = "condition"
+
+class SkyblockGuideHighlightFeature private constructor(
+    private val config: () -> Boolean,
+    inventory: RepoPattern,
+    loreCondition: RepoPattern,
+    private val onSlotClicked: (GuiContainerEvent.SlotClickEvent) -> Unit = {},
+    private val onTooltip: (LorenzToolTipEvent) -> Unit = {},
+) {
+
+    private val inventoryPattern by inventory
+    private val conditionPattern by loreCondition
+
+    private constructor(
+        config: () -> Boolean,
+        key: String,
+        @Language("RegExp")
+        inventory: String,
+        @Language("RegExp")
+        loreCondition: String,
+        onSlotClicked: (GuiContainerEvent.SlotClickEvent) -> Unit = {},
+        onTooltip: (LorenzToolTipEvent) -> Unit = {},
+    ) : this(config, group.pattern("$key.$keyPrefixInventory", inventory), group.pattern("$key.$keyPrefixCondition", loreCondition), onSlotClicked, onTooltip)
+
+    private constructor(
+        config: () -> Boolean,
+        key: String,
+        @Language("RegExp")
+        inventory: String,
+        loreCondition: RepoPattern,
+        onSlotClicked: (GuiContainerEvent.SlotClickEvent) -> Unit = {},
+        onTooltip: (LorenzToolTipEvent) -> Unit = {},
+    ) : this(config, group.pattern("$key.$keyPrefixInventory", inventory), loreCondition, onSlotClicked, onTooltip)
```

Comment:
are you sure that repo pattern will work in this case? other instances where enums use the repo pattern inizialize every single enum instance in skyhanni main class on load event manually

Reply:
it works when called in the init of an object which is loaded. eg. the companion object